### ISSUE TITLE
Using malloc_usable_size() in arena triggers compactions early and results in smaller files and diff number of files

### ISSUE THREAD

**Issue by yuslepukhin** [facebook/rocksdb#4252]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://www.facebook.com/groups/rocksdb.dev

### Expected behavior

Some tests such as column_family_test,db_compaction_test rely on producing a specific number of SST files in directories. To that end they employ a `SpecialMemTable` rep that causes flushes to take place once once a certain threshold is reached. 

### Actual behavior

However, the compaction is triggered based on the amount of memory used within the `Memtable`. Some debug and non-debug allocators have a higher allocation overhead and the `malloc_usable_size` reported is higher. This results in earlier flushes, smaller files, smaller sizes per level or db path. 

Some tests such as `db_compaction_test` and `column_family_test` expect certain number of files to be created as a result of inserting data. This creates a erroneous connection between the amount data inserted and the amount of memory allocated within the `Memtable`, and this results in smaller amount of data inserted per file, the number of files is higher and the tests fail.

IMHO, either the number

**Comment by anand1976** [comment]
@yuslepukhin The db_compaction_test and column_family_test are using ```SpecialMemTableRep```, which triggers flush when the configured number of entries is reached. This should not depend on malloc_usable_size(). Do you have an example of the test failure?

**Comment by yuslepukhin** [comment]
@anand1976 Yes, I do. The flush takes place earlier than the configured number of entries. The end result is that I have 4 SST files and not 3 as the test expects. The moment I disable malloc_usable_size() within the ConcurrentArena, things start working as expected in `db_compaction_test`.
`column_family_test` is yet another story, where the overhead is set to 1.1 and it simply does not work for jemalloc. `_msize` works though.

**Comment by yuslepukhin** [comment]
@anand1976 I will follow up as soon as I can.

**Comment by yuslepukhin** [comment]
@anand1976 In reality, all you need to do is to increase the number returned by malloc_usable_size() by 25% at that location and it should repro.

**Comment by anand1976** [comment]
@yuslepukhin The problem seems to be with ```write_buffer_size```. When this is set, ```WriteBufferManager``` is enabled and its calculation of memory usage is determined by ```malloc_usable_size()```, as that's the number that the arena allocator uses when it calls ```WriteBufferManager::ReserveMem```.

**Comment by yuslepukhin** [comment]
Correct, so the tests assume that the allocated size is the same as data size. Previously it was close enough, but after this change to usable size the difference grew.
